src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Internal.hs

```diff
@@ -98,12 +120,20 @@ witnessGenerator circuit inputs =
     in
         result
 
+indexW :: Representable i => ArithmeticCircuit a i o -> i a -> Var a i -> a
```

Comment:
Note for future: maybe we can create a helper datatype like this:
```haskell
data Witness a i = Witness { wInput :: i a, wNew :: Map Natural a }
instance Representable i => Representable (Witness a i) where
  type Rep (Witness a i) = Var a i
```
This would reduce clutter around witnesses a little.

Reply:
I don't think this is `Representable` because it's not fixed-size/shape, even though `i` is. In particular `Map Natural` is not `Representable`.

Reply:
Oh yeah, sorry
I was thinking abt creating the empty map in `tabulate`, but I guess it doesn't satisfy the laws of `Representable`